cmd/codegen/generator/go/templates/functions.go

```diff
@@ -89,6 +89,29 @@ func (funcs goTemplateFuncs) FuncMap() template.FuncMap {
 	}
 }
 
+func (funcs goTemplateFuncs) FormatReturnType(f introspection.Field, scopes ...string) (representation string, err error) {
+	scope := strings.Join(scopes, "")
+	ff := funcs.CommonFunctions.FormatTypeFuncs.WithScope(scope)
+
+	for ref := f.TypeRef; ref != nil; ref = ref.OfType {
+		switch ref.Kind {
+		case introspection.TypeKindNonNull:
+			continue
+		case introspection.TypeKindList:
+			defer func() {
+				representation = ff.FormatKindList(representation)
+			}()
+		case introspection.TypeKindObject:
+			representation = ff.FormatKindObject(representation, ref.Name, funcs.CommonFunctions.ConvertID(f))
+			representation = "*" + representation
+			return
+		default:
+			return funcs.CommonFunctions.FormatType(ref, scope, funcs.CommonFunctions.ConvertID(f))
+		}
+	}
+	return "", fmt.Errorf("unexpected type kind %s", f.TypeRef.Kind)
+}
```

Comment:
This is quite a bit hack. Feel free to suggest if anyone has a better solution. 🙏

Reply:
I think it's fine that way.

A big refactor on how the codegen works will happen at some point and that will be a good time to fix all these small magic tricks.